vumi/transports/wechat/wechat.py

```diff
@@ -473,13 +491,16 @@ def push_message(self, wc_message, vumi_message):
                 lambda ack: self.clear_addr_mask(wc_message.from_user_name))
         return d
 
+    @inlineCallbacks
     def handle_api_response(self, response, message):
         if http_ok(response):
-            return self.publish_ack(user_message_id=message['message_id'],
-                                    sent_message_id=message['message_id'])
-        return self.publish_nack(
+            ack = yield self.publish_ack(user_message_id=message['message_id'],
+                                         sent_message_id=message['message_id'])
+            returnValue(ack)
+        nack = yield self.publish_nack(
             message['message_id'],
             reason='Received status code: %s' % (response.code,))
+        returnValue(nack)
```

Comment:
Can't remember tbh, maybe because we were doing other things in this method at some stage (like publishing statuses). I think we can put it back to how it was.